docs/elements/reference/common.mdx

```diff
@@ -0,0 +1,444 @@
+| `children` | `(args: { state: 'success' \| 'error' \| 'warning' \| 'info' \| 'idle'; message: string \| undefined; codes: ErrorMessageKey[] \| undefined }) => React.ReactNode` | Use this function to access the field's state. Optionally, information regarding password validation is given. |
+
+### Examples {{ toc: false }}
+
+#### Basic
+
+Initially, the `<Input>` will have a `state` of `idle` until the user interacts with the input. Depending on the user's input, the `state` will change to `success` or `error`.
+
+```tsx {4-6}
+<Clerk.Field name="emailAddress">
+  <Clerk.Label>Email</Clerk.Label>
+  <Clerk.Input />
+  <Clerk.FieldState>
+    {({ state }) => <p>Field's state is: {state}</p>}
+  </Clerk.FieldState>
+</Clerk.Field>
+```
+
+#### `<Input type="password" validatePassword>`
+
+If you're using [`<Input type="password" validatePassword>`](#input-type-password) the `<FieldState>`'s children function receives additional arguments: `codes` and `message`.
+
+Initially, the `<Input>` will have a `state` of `idle` until the user interacts with the input. Depending on the user's input, the `state` will change to `'success' | 'error' | 'warning' | 'info'`.
+
+`message` is the english, standardized message generated for the current state of the input. The `message` is generated through `codes`. By having access to `codes` yourself you can customize or localize the returned `message`.
+
+```tsx {4-12}
+<Clerk.Field name="password">
+  <Clerk.Label>Password</Clerk.Label>
+  <Clerk.Input type="password" validatePassword />
+  <Clerk.FieldState>
+    {({ state, codes, message }) => (
+      <div>
+        <pre>Field state: {state}</pre>
+        <pre>Field msg: {message}</pre>
+        <pre>Codes: {codes?.join(', ')}</pre>
+      </div>
+    )}
+  </Clerk.FieldState>
+</Clerk.Field>
+```
+
+## `<GlobalError />`
+
+Render errors that are returned from Clerk's API, but that are not associated with a specific form field. By default, will render the error's message wrapped in a `<div>`. Optionally, the `children` prop accepts a function to completely customize rendering. Must be placed **inside** components like `<SignIn>`/`<SignUp>` to have access to the underlying form state.
+
+### Properties {{ toc: false }}
+
+| Name       | Type                                                                | Description                                                                                 |
+| ---------- | ------------------------------------------------------------------- | ------------------------------------------------------------------------------------------- |
+| `asChild` | `boolean` (optional) | If `true`, `<GlobalError>` will render as its child element, passing along any necessary props. Default: `false` |
+| `code` | `string` (optional) | Forces the message with the matching code to be shown. This is useful when using server-side validation. Default: `undefined` |
```

Comment:
Do we have a list of the possible `code` values that we can link to? Is it just HTTP status codes?

Reply:
This is definitely something we should add, at the moment we need to pull it from our backend API. Let's follow up with the concrete values 👍